packages/@lwc/engine-dom/src/customElements.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { create, isFunction, setPrototypeOf } from '@lwc/shared';
+import { patchCustomElementRegistry } from './patches/global-registry';
+
+export type UpgradeCallback = (elm: HTMLElement) => void;
+export interface UpgradableCustomElementConstructor extends CustomElementConstructor {
+    new (upgradeCallback: UpgradeCallback): HTMLElement;
+}
+
+let HTMLElementConstructor;
+let getUpgradableElement: (tagName: string) => CustomElementConstructor;
+let getUserConstructor: (
+    upgradeCallback: UpgradeCallback
+) => UpgradableCustomElementConstructor | UpgradeCallback;
+
+function isCustomElementRegistryAvailable() {
+    if (typeof customElements === 'undefined') {
+        return false;
+    }
+    try {
+        // dereference HTMLElement global because babel wraps globals in compat mode with a
+        // _wrapNativeSuper()
+        // This is a problem because LWCUpgradableElement extends renderer.HTMLElement which does not
+        // get wrapped by babel.
+        const HTMLElementAlias = HTMLElement;
+        // In case we use compat mode with a modern browser, the compat mode transformation
+        // invokes the DOM api with an .apply() or .call() to initialize any DOM api sub-classing,
+        // which are not equipped to be initialized that way.
+        class clazz extends HTMLElementAlias {}
+        customElements.define('lwc-test-' + Math.floor(Math.random() * 1000000), clazz);
+        new clazz();
+        return true;
+    } catch {
+        return false;
+    }
+}
+
+if (isCustomElementRegistryAvailable()) {
+    const definePivotCustomElement = patchCustomElementRegistry();
+
+    // It's important to cache window.HTMLElement immediately after calling patchCustomElementRegistry().
+    // Otherwise, someone else could overwrite window.HTMLElement (e.g. another copy of the engine, or another pivot
+    // implementation) and we would get "Illegal constructor" errors because the HTMLElement prototypes are mixed up.
+    const { HTMLElement } = window;
```

Comment:
@caridy You might like this one... it turned out that the `Illegal constructor` error we were seeing in Jest was a true bug. This is the fix ^

The repro is to load two copies of the LWC engine, and then create an element with the _old_ LWC engine (underneath the new one). The old LWC would grab the most recent `window.HTMLElement`, which actually came from the _new_ pivot registry, and then try to mix it with the `window.HTMLElement` from the _old_ pivot registry, causing all sorts of chaos. 😛 

And here is the test to repro:

https://github.com/salesforce/lwc/blob/e1c5f816eb13e8f2d8943541f3e40f664a6938ac/packages/%40lwc/integration-karma/test/custom-elements-registry/index.spec.js#L161-L172